src/grpc/server.rs

```diff
@@ -31,19 +33,28 @@ impl ConsumerService for TaskbrokerServer {
 
         match inflight {
             Ok(Some(inflight)) => {
+                let result = self.store.get_activation_blob(&inflight.id).await;
+                let activation = match result {
+                    Ok(activation) => activation,
+                    Err(e) => {
+                        error!("Unable to retrieve pending activation: {:?}", e);
+                        return Err(Status::internal("Unable to retrieve pending activation"));
+                    }
+                };
                 let now = Utc::now();
-                let received_to_gettask_latency = inflight.received_latency(now);
+                let received_to_gettask_latency =
+                    received_latency(activation.clone(), inflight.delay_until, now);
```

Comment:
Having to clone here could be expensive for larger activation payloads. I don't think that has to block sandbox testing though.